haskell/signable-haskell-protoc/src/Data/ProtoLens/Compiler/ModuleName.hs

```diff
@@ -0,0 +1,24 @@
+module Data.ProtoLens.Compiler.ModuleName
+    ( protoModuleName ) where
+
+import Data.Char (toUpper)
+import Data.List (intercalate)
+import System.FilePath
+
+-- | Get the Haskell module name corresponding to a given .proto file.
+protoModuleName :: FilePath -> String
+protoModuleName path = fixModuleName rawModuleName
+  where
+    fixModuleName "" = ""
+    -- Characters allowed in Bazel filenames but not in module names:
+    fixModuleName ('.':c:cs) = '.' : toUpper c : fixModuleName cs
+    fixModuleName ('_':c:cs) = toUpper c : fixModuleName cs
+    fixModuleName ('-':c:cs) = toUpper c : fixModuleName cs
```

Comment:
This code block is wrong in case where "c" is a special symbol as well.

Reply:
i guess it is now used anymore. will remove it

Reply:
hm. it used but https://github.com/google/proto-lens/blob/master/proto-lens-protoc/src/Data/ProtoLens/Compiler/ModuleName.hs it is done this way in upstream

Reply:
Hmm, actually this module is exposed by proto-lens-protoc. Why do we copy this? Let's use it from proto-lens-protoc package if we need it. DRY etc..